packages/ethereum-swap-provider/lib/EthereumSwapProvider.js

```diff
@@ -70,31 +70,48 @@ export default class EthereumSwapProvider extends Provider {
       '60', '48', // PUSH1 48
       'f1', // CALL
 
+      // Validate input size
+      '36', // CALLDATASIZE
+      '60', '20', // PUSH1 20 (32 bytes)
+      '14', // EQ
+      '16', // AND (to make sure SHA256 CALL succeeded)
+
       // Validate with secretHash
       '7f', secretHash, // PUSH32 {secretHashEncoded}
       '60', '21', // PUSH1 21
       '51', // MLOAD
       '14', // EQ
-      '16', // AND (to make sure CALL succeeded)
+      '16', // AND (to make sure input valid size)
       // Redeem if secret is valid
-      '60', '47', // PUSH1 {redeemDestination}
+      '60', '4c', // PUSH1 {redeemDestination}
       '57', // JUMPI
 
       // Check time lock
       '64', expirationEncoded, // PUSH5 {expirationEncoded}
       '42', // TIMESTAMP
       '11', // GT
       // Refund if timelock passed
-      '60', '5e', // PUSH1 {refundDestination}
+      '60', '89', // PUSH1 {refundDestination}
       '57',
 
       'fe', // INVALID
 
       '5b', // JUMPDEST
+      // emit Claim(bytes32 _secret)
+      '7f', '8c1d64e3bd87387709175b9ef4e7a1d7a8364559fc0e2ad9d77953909a0d1eb3', // PUSH32 topic Keccak-256(Refund())
```

Comment:
Comment says `Keccak-256(Refund())` in claim section

Reply:
Ah sorry yes will update